pip/utils/temp_dir.py

```diff
@@ -6,37 +6,40 @@
 from pip.utils import rmtree
 
 
-class BuildDirectory(object):
+class TempDirectory(object):
+    """Helper class that owns and cleans up a temporary directory.
+    """
```

Comment:
Maybe add a bit more detail here, including how to use the class? See my comment below on the ```cleanup()``` method for a case where I can't quite see how I'd correctly use the API of this class...